src/lib/blkalloc/append_blk_allocator.cpp

```diff
@@ -179,6 +175,12 @@ std::string AppendBlkAllocator::to_string() const {
     return fmt::format("{}, last_append_offset: {}", get_name(), m_last_append_offset);
 }
 
+blk_num_t AppendBlkAllocator::available_blks() const { return get_total_blks() - get_used_blks(); }
+
+blk_num_t AppendBlkAllocator::get_used_blks() const { return m_last_append_offset; }
+
 blk_num_t AppendBlkAllocator::get_freeable_nblks() const { return m_freeable_nblks; }
 
+blk_num_t AppendBlkAllocator::get_defrag_nblks() const { return get_freeable_nblks() - available_blks(); }
```

Comment:
actually I am little confuse about the definition of defrag_nblks() and why it use free_blks - available_blks ?
from HO GC perspective,  the chunk with larger free blks is more worth to do the compaction, right? why need to minus available blks

Reply:
i think this is should be `available_blks()- get_freeable_nblks()` = `get_total_blks() - get_used_blks()- get_used_blks()` , indicating the live data on the chunk

Reply:
if i am right , defrag_nblks() means the actual deleted number of blocks, which can be gc.
`m_freeable_nblks` means the theorically freeable number of blks, which will be updated in every `async_free`.

`available_blks()` means the actuall freeable number of blks, which means the left space never be allocated .

so the `defrag_nblks()` is ` theorically freeable` minus `actual freeable`, that is the blks need to be GC.

pls correct me if i am wrong, @yamingk

Reply:
ok, m_freeable_nblks will be initialize to all blk num of this chunk during format and will be increase during every async_free(). it is much clear to me. @JacksonYao287 @yamingk .